execution/test_utils.go

```diff
@@ -228,13 +229,49 @@ func AreStreamsEqualNoOrderingWithRetractionReduction(ctx context.Context, state
 	firstMultiSet := newMultiSet(config.Equality)
 	secondMultiSet := newMultiSet(config.Equality)
 
-	log.Println("first stream")
-	firstRecords, err := ReadAll(ctx, stateStorage, first)
+	log.Println("got stream")
+	gotRecords, err := ReadAll(ctx, stateStorage, got)
 	if err != nil {
-		return errors.Wrap(err, "couldn't read first streams records")
+		return errors.Wrap(err, "couldn't read got streams records")
+	}
+	if len(gotRecords) == 0 {
+		return nil
 	}
 
-	for _, rec := range firstRecords {
+	seenIDs := make(map[int64]struct{})
+	var wantID string
+
+	for i, rec := range gotRecords {
+		// All recordIDs should be of form X.index, where X is the same streamID for every record and indexes are distinct
+		// Store the first streamID prefix from the record
+		if i == 0 {
+			wantID, err = getBaseID(rec.ID())
+			if err != nil {
+				return errors.Wrap(err, "couldn't get base ID from first record")
+			}
+		}
+
+		// The ID should be of form streamID.index, first we want to compare streamIDs and then check if indexes are unique
+		recordIDSplit := strings.Split(rec.ID().ID, ".")
```

Comment:
This is basically the same code as in `getBaseID` - is it possible to extract this to other function (that returns `recordIDSplit[1]` for example) or at least include it in `else` case above? (So that case `i==0` doesn't execute this piece of code 2 times)